### PR TITLE
Coin Change, Subset Sum and Rod Cutting in Go

### LINKED ISSUE #1099: Adding Coin Change, Subset Sum and Rod Cutting in Go

### PR DESCRIPTION
**Pull request by singh-shreya6**
Fixes #1099 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added Solutions to classical DP problems in Golang

#### Languages Used:
- Go

#### Files Added:
- Coin_Change.go
- Subset_Sum.go
- Dynamic_Programming_Rod_Cutting.go

### CODE REVIEW

File: Coin_Change/Coin_Change.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    // Base case when value is 0
+    dp[0] = 1; 
+    
+    /*
+        The outer loop runs over the coins array. It selects
+        a coin and checks whether a given value can be made 
+        using that denomination or not. The inner loop runs
+        upto the value we want to obtain.
+    */
+    for i := 0; i<m; i++ {
```
Comment (MastersAbh) [inline Coin_Change/Coin_Change.go]: space around <

File: Coin_Change/Coin_Change.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import "fmt"
+
+// Function to calculate the number of ways of making
+// the desired change using the given coins
+func CoinChange(coins[] int, m int, value int) int {
+    dp := make([]int , value + 1) 
+
+    // Base case when value is 0
+    dp[0] = 1;
```
Comment (MastersAbh) [inline Coin_Change/Coin_Change.go]: Remove semi-colon. Not required in Go

File: Dynamic_Programming_Rod_Cutting/Dynamic_Programming_Rod_Cutting.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    pieces and sell those pieces.
+*/
+
+package main
+
+import "fmt"
+
+// Function to compute maximum of two numbers
+func max(i int, j int) int {
+    if i>j {
```
Comment (MastersAbh) [inline Dynamic_Programming_Rod_Cutting/Dynamic_Programming_Rod_Cutting.go]: spaces around >

File: Subset_Sum/Subset_Sum.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import "fmt"
+
+// Function to compute whether a subset of array exists
+// having the given sum
+func subsetSum(arr[] int, size int, sum int) bool {
+  
+    // A 2D Matrix containing boolean values, true and false
+    dp := make([][]bool, size + 1)
+
+    for i:=0; i<=size; i++ {
```
Comment (MastersAbh) [inline Subset_Sum/Subset_Sum.go]: spaces around := and <=

File: Subset_Sum/Subset_Sum.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        dp[i] = make([]bool, sum + 1)
+    }
+    
+    // We cannot form a sum i by taking no elements
+    for i:=0; i<=sum; i++ {
```
Comment (MastersAbh) [inline Subset_Sum/Subset_Sum.go]: spaces around operators

File: Subset_Sum/Subset_Sum.go [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        dp[0][i] = false
+    }
+
+    // We can always form the sum 0
+    // by taking 0 instance of the element        
+    for i:=0; i <= size; i++ {
```
Comment (MastersAbh) [inline Subset_Sum/Subset_Sum.go]: spaces around :=

Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.go]: Done @MastersAbh 
Please review

**Pull request merged**